src/mme-app/parse_config.cpp

```diff
@@ -0,0 +1,254 @@
+                log_msg(LOG_INFO, "Configured apn %s\n", apnitr->name.GetString()); 
+#endif
+                std::string apn = apnitr->name.GetString();
+                std::string spgw = apnitr->value.GetString();
+
+                log_msg(LOG_INFO, "Configured service %s\n", spgw.c_str());
+                log_msg(LOG_INFO, "Configured apn %s \n", apn.c_str()); 
+                // add apn==>{ spgw service,  address, failures }  in the map 
+                apn_config *ap1 = new apn_config(apn, spgw);
+                mme_tables->add_apn(ap1);
+            }
+        }
+    }
+    /* Print parsed configuraton */
+    log_msg(LOG_DEBUG,"mme_name : %s \n", config->mme_name);
+    log_msg(LOG_DEBUG,"mme_groupid : %d \n", config->mme_group_id);
+    log_msg(LOG_DEBUG,"mme_code : %d \n", config->mme_code);
+    log_msg(LOG_DEBUG,"logging  : %s \n", config->logging);
+    log_msg(LOG_DEBUG,"MCC :    %d %d %d \n", config->mcc_dig1, config->mcc_dig2, config->mcc_dig3);
+    log_msg(LOG_DEBUG,"MNC :    %d %d %d \n", config->mnc_dig1, config->mnc_dig2, config->mnc_dig3);
+    for(int i=0; i<config->num_plmns;i++)
+    {
+        log_msg(LOG_DEBUG,"PLMN(%d) :    %d %d %d \n", i, config->plmns[i].idx[0], config->plmns[i].idx[1], config->plmns[i].idx[2]);
+    }
+    mme_tables->initiate_spgw_resolution();
+    return ;
+}
+
+/* One resolution at a time with timer retry */
+void mmeConfig::initiate_spgw_resolution()
+{
+    std::list<apn_config*>::iterator it;
+    bool started = false;
+    apn_config *temp;
+
+    for(it = apn_list.begin(); it != apn_list.end(); it++) 
+    {
+        temp = *it;
+        if(temp->get_dns_state() == false)
+        {
+            struct in_addr p = {0};
+            struct in_addr s = {0};
+            p.s_addr = temp->get_pgw_addr();
+            s.s_addr = temp->get_sgw_addr();
+            log_msg(LOG_DEBUG, "SGW address = %s, PGW address = %s ",inet_ntoa(p), inet_ntoa(s));
+            continue;
+        }
+        struct addrinfo hints;
+        struct addrinfo *result=NULL, *rp=NULL; 
+        int err;
+
+        memset(&hints, 0, sizeof(struct addrinfo));
+        hints.ai_family = AF_INET;    /* Allow IPv4 or IPv6 */
+        hints.ai_socktype = SOCK_DGRAM; /* Datagram socket */
+        hints.ai_flags = AI_PASSIVE;    /* For wildcard IP address */
+        hints.ai_protocol = 0;          /* Any protocol */
+        hints.ai_canonname = NULL;
+        hints.ai_addr = NULL;
+        hints.ai_next = NULL;
+        err = getaddrinfo(temp->get_spgw_srv().c_str(), NULL, &hints, &result);
+        if (err != 0) 
+        {
+            // Keep trying ...May be SGW is not yet deployed 
+            // We shall be doing this once timer library is integrated 
+            log_msg(LOG_ERROR, "%s - getaddr info failed %s\n",temp->get_spgw_srv().c_str(), gai_strerror(err));
+            if(started == false)
+              MmeTimerUtils::startTimer(10000, 1 /* ue_index */, 1 /*timer type*/, mmeConfigDnsResolve_c);
```

Comment:
startTimer assumes Ue context timer.
Here also, we may need a different api for starting global timers? TimerContext can be used instead of MmeUeTimerCtxt.
TimerType = mmeConfigDnsResolve_c. TimerId = 0.

Reply:
Yeah...since code was commented and not used, i added if loop in the callback. Thought thats good starting point. 
Based on the timerId we can use ue_index or skip ue_index...   

Or

Each module can call the timer APIs independently.